File: src/app/movie.ts
import { FileHandleModel } from "./file-handle-model";

export interface Movie {
    moviePK: MoviePk
    noOfTickets: number
    movieGenre: string
    movieHours: string
    movieLanguage: string
    movieDescription: string
    movieRating: string
    movieDate: any
    ticketStatus: boolean
    movieImage: any[]
  }
  
  export interface MoviePk {
    movieName: string
    theatreName: string
  }
  

File: moviebookingapp/src/main/java/com/moviebookingapp/model/Movie.java
package com.moviebookingapp.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document("movie")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Movie {

    @Id
    private MoviePK moviePK;
    private int noOfTickets;
    private String movieGenre;
    private String movieHours;
    private String movieLanguage;
    private String movieDescription;
    private String movieRating;
    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate movieDate;
    private boolean ticketStatus;
    private ImageModel movieImage;
}
